impl/libs/librefos/src/refos-rpc/rpc.c

```diff
@@ -143,14 +143,14 @@ rpc_init(const char* name_str, int32_t label)
 
 	rpc_reset_contents(NULL);
 
-    seL4_SetMR(0, label);
-
     if (!_rpc_recv_cslot) {
         rpc_setup_recv(REFOS_THREAD_CAP_RECV);
     } else if (seL4_MessageInfo_get_extraCaps(_rpc_minfo) > 0) {
         // Flush recieving path of previous recieved caps.
         seL4_CNode_Delete(REFOS_CSPACE, _rpc_recv_cslot, REFOS_CDEPTH);
     }
+
+    seL4_SetMR(0, label);
```

Comment:
when doing a system call, the second system call was always being corrupted
I surmised that in the else block, the seL4_CNode_Delete was somehow deleting the label which was put in the seL4_SetMR call.

thanks for calling this out as perhaps it needs a different fix.